src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -158,7 +158,18 @@ public class JsonInputFormatter : TextInputFormatter
 
                     if (successful)
                     {
-                        return InputFormatterResult.SuccessAsync(model);
+                        if (model == null && !context.AllowEmptyInput)
```

Comment:
I think @SteveSandersonMS 's point about reasonably expected behavior about what is "empty data" and what `IsValid` should return makes sense here, so let's go with that.